SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -50,13 +62,13 @@ bool ComponentFactory::FillSchemaObject(Schema_Object* ComponentObject, UObject*
 			const FRepParentCmd& Parent = Changes.RepLayout.Parents[Cmd.ParentIndex];
 
 #if TRACE_LIB_ACTIVE
-			if (LatencyTracer != nullptr)
+			if (LatencyTracer != nullptr && OutLatencyTraceId != nullptr)
 			{
-				OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
-				if (OutLatencyTraceId == USpatialLatencyTracer::InvalidTraceKey)
+				*OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
+				if (*OutLatencyTraceId == USpatialLatencyTracer::InvalidTraceKey)
 				{
 					// Check for sending a nested property
-					OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
+					*OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
```

Comment:
As I understand, `OutLatencyTraceId` will be equal to the last value we process in that loop? All the previous values are discarded?

Reply:
Yeah that's true, I will add some code to end traces which are about to be overwritten - we can only send 1 trace with the Schema object at a time. Perhaps I could extend it to send more!

Reply:
If this issue is not resolved in this PR - could you create a jira ticket and add a `// TODO: UNR-<ticket number>` with a short description here and in the case below?

Reply:
We're going to terminate the traces except one in this situation, as it's a bit tricky to manage sending multiple traces!

Reply:
If that's going to be the intentional behaviour - could you put a comment that says so?